api/database_cluster_backup.go

```diff
@@ -67,11 +74,34 @@ func (e *EverestServer) CreateDatabaseClusterBackup(ctx echo.Context, namespace
 }
 
 // DeleteDatabaseClusterBackup deletes the specified cluster backup on the specified kubernetes cluster.
-func (e *EverestServer) DeleteDatabaseClusterBackup(ctx echo.Context, namespace, name string) error {
+func (e *EverestServer) DeleteDatabaseClusterBackup(
+	ctx echo.Context,
+	namespace, name string,
+	params DeleteDatabaseClusterBackupParams,
+) error {
+	cleanupStorage := pointer.Get(params.CleanupBackupStorage)
+	// If cleanup is required, we add a finalizer to the backup object.
+	if cleanupStorage {
+		reqCtx := ctx.Request().Context()
+		backup, err := e.kubeClient.GetDatabaseClusterBackup(reqCtx, namespace, name)
+		if err != nil {
+			return err
+		}
+		if err := e.cleanupBackupStorage(reqCtx, backup); err != nil {
```

Comment:
Shall the backups be removed only after the DB has been successfully removed?

Reply:
No, its actually the other way around:

- Add this finalizer on all backups for the DB
- DB deletion is triggered, however there is a finalizer set on it by everest-operator
- DB finalizer will delete all the DBBackups (everest-operator)
- DBBackup finalizer is handled (storage clean-up)
- DB is GC-ed by k8s only after all backups are gone